### ISSUE TITLE
Documentation needed on valid values for various az commands

### ISSUE THREAD

**Issue by srikcgaa2** [Azure/azure-cli#23097]
## Describe the bug

**Command Name**
`az billing period show`

**Errors:**
```
argument --name/-n: expected one argument

```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- `az billing period show --name`

## Environment Summary
```
Windows-10-10.0.14393-SP0
Python 3.10.4
Installer: MSI

azure-cli 2.37.0

Extensions:
account 0.2.3
azure-devops 0.18.0
azure-iot 0.12.1

Dependencies:
msal 1.18.0b1
azure-mgmt-resource 21.1.0b1
```
## Additional Context
In this command, there is no documentation that shows where I can get the values for the "name" parameter.  I tried using the "az billing period list" and it gave me empty list.  I have logged in to Azure and set the subscription as well because I am able to get all the key vaults in the same powershell window.

Similar issue exists (on missing documentation) for many other commands and arguments as well.

**Comment by yonzhan** [comment]
route to CXP team

**Comment by navba-MSFT** [comment]
@srikcgaa2 Thanks for reaching out to us and reporting this issue. We are looking into this issue and we will provide an update.

**Comment by navba-MSFT** [comment]
@srikcgaa2 `az billing period show` and `az billing period list` commands are only supported for Azure Web-Direct subscriptions. Other subscription types which were not purchased directly through the Azure web portal are not supported through this preview API. More Info [here](https://docs.microsoft.com/en-us/cli/azure/billing/period?view=azure-cli-latest#az-billing-period-list).
I was able to run the command and get the available period list name: 
**az billing period list**
>{
  "billingPeriodEndDate": "2022-07-31",
  "billingPeriodStartDate": "2022-07-01",
  "id": "/subscriptions/b83c1ed3-c5b6-44fb-b5ba-2b83a074c23f/providers/Microsoft.Billing/billingPeriods/202209-1",
  "invoiceIds": null,
  **"name": "202209-1",**
  "type": "Microsoft.Billing/BillingPeriods"
}

I used the above name to fetch my named billing period as shown below:
**az billing period show -n 202209-01**

I will create a PR to improvise our documentation. Hope this helps.
